### DISCUSSION TITLE
Specifying number density for injectors from data file

### DISCUSSION THREAD

**Discussion by cooluser26** [epochpic/epoch#585] [General]
Hello,

I'm new to using the injector block, and I have managed to load data files into the injector block to get a drift profile. However,  I'm unsure how to implement a number density profile or number. It is possible to do it by controlling the weight of the particles. My question is, do all the injected particles get injected into the first cell nearest to the x_min boundary? 

e.g in 1D if we have n_x = 6000 and x_max = 600e-6 then dx = 1e-7 --> dV = 1e-7 m^3 so if I want a density of 1e+20 m^-3 then I would need 1e+13 real particles. From this, I can set the weight based on the number of macro-particles.

Would this be the right way to do it? Thanks a lot!

William.

**Comment by Status-Mirror** [comment]
Hey William,

I'm guessing you're using files to specify the properties of a plasma injector, and that you don't have `inject_from_file = T` in your injector block.

The EPOCH simulation grid is surrounded by ghost cells. When you use a plasma injector, ghost cells immediately outside the simulation window are filled with macro-particles describing your desired distribution - they load the requested particles per cell, with weights and momentum distributions to match the density and temperature set by the injector block. If you set number density $n$ in a cell of volume $V$, and choose $n_{ppc}$ particles per cell, then the loaded macro-particle weights will be:

$W = nV/n_{ppc}$

These macro-particles move one step using the standard EPOCH particle pusher, and any macro-particles which enter the simulation window are saved - the rest are deleted. Hence, macro-particles are injected into the simulation as if the plasma continues beyond the simulation boundary.

Hope this helps,
Stuart
